core/controller/src/main/scala/whisk/core/loadBalancer/LoadBalancerService.scala

```diff
@@ -339,7 +342,23 @@ class LoadBalancerService(config: WhiskConfig, instance: InstanceId, entityStore
   }
 }
 
-object LoadBalancerService {
+/**
+ * An Spi for providing LoadBalancerService implementations
+ */
+trait LoadBalancerProvider extends Spi {
+  def requiredProperties: Map[String, String]
+
+  def loadBalancer(whiskConfig: WhiskConfig, instance: InstanceId, entityStore: EntityStore)(
```

Comment:
I don't think `entityStore` should be part of this interface, it is an impl detail of the invokerpool, afaik.

Reply:
The `entityStore` is here only to create the health check. I think this can be separated from the load balancer service. The load balancer doesn't need access to the `entityStore`. We should factor it out.

This creates an ordering issue: factor it out first then introduce the SPI, or can we introduce the SPI without the entity store but have a concrete implementation that used the entity store to preserve existing behavior today?

Since SPI changes are breaking API changes potentially, we should avoid introducing an SPI that we'll change soon there after.